app-engine/image-resizing/main.py

```diff
@@ -56,8 +56,17 @@ def root():
             serving_url = images.get_serving_url(
                 blob_key=None, filename=blobstore_filename
             )
-        except (images.Error, urllib.error.URLError):
-            logging.error("Unable to get %s retry %s", blobstore_filename, cpt)
+        except (images.Error, urllib.error.URLError) as exc:
+            logging.info(
+                "Failed getting serving url on filename %s", blobstore_filename
+            )
```

Comment:
Je ne comprends pas l'intérêt de ce log. On a autant d'information (et même plus) dans le `logger.error` dessous.

Ma suggestion était d'ajouter un `logger.info` à un endroit où l'on passe systématiquement. En tout début de fonction, par exemple, ou juste avant la boucle `for`.

Reply:
@dbaty justement je demandais a @Vencespass de ne pas rajouter ou on passe tout le temps. 
Car on passe par cette fn à chaque fois qu'on veut get une image donc ça fait un peu, beaucoup, à la folie, du log. 

Si tu es sur de ton coup et que ça ne vas pas polluer les logs, ok.

Reply:
Je pensais que l'objectif était de tester si le `logger.info` remontait et que il était indiscutable que on passait dans le `except`. 

Est-ce que si on log 30 images par pages, que chaque utilisateur affiche en moyenne 10 pages,  qu'on a 100k utilisateurs par mois, alors on rajoute 360 millions de ligne de logs par ans. n'est-ce pas un peu exagéré xD?

Reply:
Pourquoi pas en log qu'une fraction (`random.randint(1, 10) * 1000`, par ex)

Reply:
> Je pensais que l'objectif était de tester si le logger.info remontait et que il était indiscutable que on passait dans le except.

Ajouter ce `logger.info` à cet endroit est inutile. Il ne logge rien de plus que le `logger.error` à la ligne suivante.

> Est-ce que si on log 30 images par pages, que chaque utilisateur affiche en moyenne 10 pages, qu'on a 100k utilisateurs par mois, alors on rajoute 360 millions de ligne de logs par ans. n'est-ce pas un peu exagéré xD?

C'est exagéré si on garde le `logger.info`. Mais on pourrait juste le déployer pendant quelques heures ou jours, constater qu'on a bien des logs puis supprimer la ligne.

Reply:
fyi: le support google nous suggère de faire cela : 

```py
 except (images.Error, urllib.error.URLError) as e:
…
raise Exception(f"Unable to get serving url {blobstore_filename}" + “ Exception: {e}”)
````

Je ne suis pas sur de ce qu'on attend avec ce log info, de savoir si il remonte dans gcp ? si c'est le cas, si on en prends qu'un fraction des requêtes, cela me semble mieux et remplis le même job ?

Reply:
> Je ne suis pas sur de ce qu'on attend avec ce log info, de savoir si il remonte dans gcp ?

Oui, c'est pour vérifier que toutes nos utilisations de `logger.xxxx()` sont bien visibles dans les logs GCP.

> si c'est le cas, si on en prends qu'un fraction des requêtes, cela me semble mieux et remplis le même job ?

Sûrement. Comme tu veux ! :)


> fyi: le support google nous suggère de faire cela :

Ça affichera le message d'erreur de la dernière tentative (échouée) de la boucle `for`... qu'on aura de toute façon déjà loggé via le `logger.error()` plus haut. Ça me semble inutile.